### ISSUE TITLE
Tensorflow - Disaccording TF METRICS (Accuracy, Precision, Recall)

### ISSUE THREAD

**Issue by lucadigiammarino** [tensorflow/tensorflow#23126]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04.4
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 1.10.1
- Python version: 2.7
- GCC/Compiler version (if compiling from source): 5.4.0
- CUDA/cuDNN version: release 7.5
- GPU model and memory: Tesla K80 - 11.17GiB
- Bazel version: N/A
- Exact command to reproduce: N/A
- Mobile device: N/A

I am working with a CNN for an instrument recognition problem, using **IRMAS** dataset. 

The **IRMAS** dataset has a clear **training** set, based only on singular instruments, therefore each instrument (guitar, violin, etc) has its own directory with multiple samples (*multi-classification*). The **testing** set, differently, contains polyphonic music, therefore multiple classes of instruments may appear in the same song (*multi-classification*, *multi-label*).

This is the final part of my network with the prediction and loss calculation, I am using the `Estimator API`.

    

    ### previous part of network is omitted
    
    # logits layer
    logits = tf.layers.dense(inputs=dropout, units=labels.shape[1])

    print('Shape Logits:', logits.shape)
       
   
    predictions = {
        "classes": tf.argmax(input=logits, axis=1),
        "probabilities": tf.nn.sigmoid_cross_entropy_with_logits(labels=tf.cast(labels, tf.float32), logits=tf.cast(logits, tf.float32), name="sigmoid_tensor")
    }

    if mode == tf.estimator.ModeKeys.PREDICT:
        return tf.estimator.EstimatorSpec(mode=mode, predictions=predictions)

    loss = tf.losses.sigmoid_cross_entropy(multi_class_labels=labels, logits=logits)

    if mode == tf.estimator.ModeKeys.TRAIN:
        optimizer = tf.train.AdamOptimizer(learning_rate=0.001)
        #optimizer = tf.train.MomentumOptimizer(learning_rate=0.01, momentum=0.96)
        train_op = optimizer.minimize(
                                  loss=loss,
                                  global_step=tf.train.get_global_step())
        logging_hook = tf.train.LoggingTensorHook({"loss" : loss}, every_n_iter=10)
        return tf.estimator.EstimatorSpec(mode=mode, loss=loss, train_op=train_op, training_hooks = [logging_hook])

    eval_metric_ops = {
        "accuracy": tf.metrics.accuracy(
        labels=tf.argmax(input=labels, axis=1),
        predictions=predictions["classes"]),

        "recall": tf.metrics.recall(
        labels=tf.argmax(input=labels, axis=1),
        predictions=predictions["classes"]),

        "precision": tf.metrics.precision(
        labels=tf.argmax(input=labels, axis=1),
        predictions=predictions["classes"])
    }

After few epochs, the training process seems to work since the loss goes down quite smoothly. However, the tensorflow metrics needed to evaluate the model does not seem to agree.

These are my results on the evaluation set (15% of the training set):

`{'recall': 0.9989496, 'accuracy': 0.40656063, 'global_step': 1428, 'precision': 0.95004994, 'loss': 0.2432195}`

These are the results obtained from the testing set:

`{'recall': 0.9981618, 'accuracy': 0.26530612, 'global_step': 1428, 'precision': 0.96533334, 'loss': 0.46097097}`

In my opinion, the accuracy value seems to be the only right, since decreases in the testing set, which is composed by polyphonic music rather than singular instruments. I don't understand why I get these results for precision and recall, they should be lower compared to the accuracy I get. 

I've tried to display the FP, FN, TN and TP and they seem to be in accord with precision and recall. In fact, by computing the accuracy with them I got around 0.97 (too high for this classification problem).

My question is why the metrics of tensorflow are so in disaccord? I could not find any useful information in the documentation.

Thank you in advance

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow-tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

**Comment by tensorflowbutler** [comment]
It has been 17 days with no activity and the `awaiting response` label was assigned. Is this still an issue?
